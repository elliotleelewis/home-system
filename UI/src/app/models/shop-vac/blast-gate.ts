export interface BlastGate {
	id: string;
	isOpen: boolean;
	createdAt: Date;
	updatedAt: Date;
}
