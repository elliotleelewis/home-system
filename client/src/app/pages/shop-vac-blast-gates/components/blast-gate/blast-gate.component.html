<div class="level">
	<div class="level-left">
		<input
			id="blast-gate-{{ blastGate.id }}"
			class="switch"
			type="checkbox"
			name="is-open"
			[ngModel]="blastGate.isOpen"
			(change)="handleToggleBlastGate(blastGate)"
		/>
		<label for="blast-gate-{{ blastGate.id }}">{{ blastGate.name }}</label>
	</div>
	<div class="level-right">
		<button
			class="button is-warning"
			type="button"
			(click)="handleActivateBlastGate(blastGate)"
		>
			Activate
		</button>
		<div class="dropdown is-hoverable is-right ml-2">
			<div class="dropdown-trigger">
				<button class="button is-default">
					<span class="icon"><i class="fas fa-ellipsis-v"></i></span>
				</button>
			</div>
			<div class="dropdown-menu" id="dropdown-menu" role="menu">
				<div class="dropdown-content">
					<a
						href="javascript:void(0)"
						role="button"
						class="dropdown-item"
						(click)="handleDeleteBlastGate(blastGate)"
					>
						<span class="icon"
							><i class="fas fa-trash-alt"></i></span
						>Delete
					</a>
				</div>
			</div>
		</div>
	</div>
</div>
