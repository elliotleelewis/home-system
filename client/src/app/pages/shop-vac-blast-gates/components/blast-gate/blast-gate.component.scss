.level {
	inline-size: 100%;
}
