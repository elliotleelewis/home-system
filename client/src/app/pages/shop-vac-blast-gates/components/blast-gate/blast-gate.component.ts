import {
	ChangeDetectionStrategy,
	Component,
	EventEmitter,
	Input,
	Output,
} from '@angular/core';
import { FormsModule } from '@angular/forms';
import { type BlastGate } from '@app/schema';

@Component({
	selector: 'app-blast-gate',
	templateUrl: './blast-gate.component.html',
	changeDetection: ChangeDetectionStrategy.OnPush,
	imports: [FormsModule],
})
export class BlastGateComponent {
	@Input()
	blastGate!: BlastGate;

	@Output()
	readonly updateBlastGate: EventEmitter<BlastGate> =
		new EventEmitter<BlastGate>();

	@Output()
	readonly deleteBlastGate: EventEmitter<BlastGate> =
		new EventEmitter<BlastGate>();

	@Output()
	readonly activateBlastGate: EventEmitter<BlastGate> =
		new EventEmitter<BlastGate>();

	isMenuOpen = false;

	handleToggleBlastGate(blastGate: BlastGate): void {
		this.updateBlastGate.emit({
			...blastGate,
			isOpen: !blastGate.isOpen,
		});
	}

	handleDeleteBlastGate(blastGate: BlastGate): void {
		this.deleteBlastGate.emit(blastGate);
	}

	handleActivateBlastGate(blastGate: BlastGate): void {
		this.activateBlastGate.emit(blastGate);
	}

	handleSetIsMenuOpen(isMenuOpen: boolean): void {
		this.isMenuOpen = isMenuOpen;
	}
}
