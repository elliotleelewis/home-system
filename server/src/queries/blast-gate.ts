import { type BlastGate, type QueryBlastGateArgs } from '@app/schema';

import { PRISMA } from '../../prisma';

export const blastGate = async (
	_: unknown,
	{ blastGateId }: QueryBlastGateArgs,
): Promise<BlastGate> => {
	const bg = await PRISMA.blastGate.findUnique({
		where: {
			id: blastGateId,
		},
	});

	if (bg === null) {
		throw new Error(`Cannot find BlastGate with id: ${blastGateId}`);
	}

	return bg;
};
