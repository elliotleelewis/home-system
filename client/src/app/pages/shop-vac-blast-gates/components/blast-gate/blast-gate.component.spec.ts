import type { ComponentFixture } from '@angular/core/testing';
import { TestBed } from '@angular/core/testing';
import { FormsModule } from '@angular/forms';

import { BlastGateComponent } from './blast-gate.component';

describe('BlastGateComponent', () => {
	let component: BlastGateComponent;
	let fixture: ComponentFixture<BlastGateComponent>;

	beforeEach(async () => {
		await TestBed.configureTestingModule({
			declarations: [BlastGateComponent],
			imports: [FormsModule],
		}).compileComponents();
	});

	beforeEach(() => {
		fixture = TestBed.createComponent(BlastGateComponent);
		component = fixture.componentInstance;

		component.blastGate = {
			id: '12345',
			name: 'Test',
			isOpen: true,
		};

		fixture.detectChanges();
	});

	it('should create', () => {
		expect(component).toBeTruthy();
	});

	it('should render', () => {
		expect(fixture).toMatchSnapshot();
	});
});
