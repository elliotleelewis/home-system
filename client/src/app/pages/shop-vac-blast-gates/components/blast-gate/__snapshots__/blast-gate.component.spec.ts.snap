// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlastGateComponent should render 1`] = `
<app-blast-gate
  activateBlastGate={[Function EventEmitter_]}
  blastGate={[Function Object]}
  deleteBlastGate={[Function EventEmitter_]}
  isMenuOpen="false"
  updateBlastGate={[Function EventEmitter_]}
>
  <div
    class="flex items-center justify-between"
  >
    <div
      class="flex items-center gap-2"
    >
      <input
        class="toggle ng-untouched ng-pristine ng-valid"
        id="blast-gate-12345"
        name="is-open"
        type="checkbox"
      />
      <label
        for="blast-gate-12345"
      >
        Test
      </label>
    </div>
    <div
      class="flex gap-2"
    >
      <button
        class="btn btn-warning"
        type="button"
      >
         Activate 
      </button>
      <div
        class="dropdown dropdown-end"
      >
        <div
          class="btn"
          role="button"
          tabindex="0"
        >
          <i
            class="fas fa-ellipsis-v fa-fw"
          />
        </div>
        <ul
          class="menu dropdown-content z-[1] w-52 rounded-box bg-base-100 p-2 shadow"
          role="menu"
          tabindex="0"
        >
          <li>
            <a
              href="javascript:void(0)"
              role="button"
            >
              <i
                class="fas fa-trash-alt fa-fw"
              />
               Delete 
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</app-blast-gate>
`;
